cmake/FindGMP.cmake

```diff
@@ -9,17 +9,80 @@
 ## directory for licensing information.
 ##
 # Find GMP
-# GMP_FOUND - system has GMP lib
-# GMP_INCLUDE_DIR - the GMP include directory
-# GMP_LIBRARIES - Libraries needed to use GMP
+# GMP_FOUND - should always be true
+# GMP - target for the GMP library
+
+include(deps-helper)
 
 find_path(GMP_INCLUDE_DIR NAMES gmp.h gmpxx.h)
 find_library(GMP_LIBRARIES NAMES gmp)
 
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(GMP DEFAULT_MSG GMP_INCLUDE_DIR GMP_LIBRARIES)
+set(GMP_FOUND_SYSTEM FALSE)
+if(GMP_INCLUDE_DIR AND GMP_LIBRARIES)
+  set(GMP_FOUND_SYSTEM TRUE)
+
+  function(GetVersionPart OUTPUT FILENAME DESC)
+    file(STRINGS ${FILENAME} RES REGEX "^#define __GNU_MP_${DESC}[ \\t]+.*")
+    string(REGEX MATCH "[0-9]+" RES "${RES}")
+    set(${OUTPUT}
+        "${RES}"
+        PARENT_SCOPE
+    )
+  endfunction()
+  function(GetVersion OUTPUT FILENAME)
+    getversionpart(MAJOR "${FILENAME}" "VERSION")
+    getversionpart(MINOR "${FILENAME}" "VERSION_MINOR")
+    getversionpart(PATCH "${FILENAME}" "VERSION_PATCHLEVEL")
+    set(${OUTPUT}
+        "${MAJOR}.${MINOR}.${PATCH}"
+        PARENT_SCOPE
+    )
+  endfunction()
+  if(EXISTS "${GMP_INCLUDE_DIR}/gmp-x86_64.h")
+    getversion(GMP_VERSION "${GMP_INCLUDE_DIR}/gmp-x86_64.h")
```

Comment:
What systems have the `gmp-x86_64.h` header? Maybe add a comment here why it is needed? And why is it not part of the `find_path(GMP_INCLUDE_DIR NAMES gmp.h gmpxx.h)` call?

Reply:
I'm not completely sure (I copied this code from somewhere else, though I wrote it myself :-) ). IIRC, this was on CentOS...

Reply:
Until now we never had an issue with FindGMP that we had until now. Should we remove this check?